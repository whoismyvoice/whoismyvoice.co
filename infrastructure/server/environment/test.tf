resource "aws_ssm_parameter" "test_mixpanel_token" {
  name        = "/test/mixpanel_token"
  description = "Token for the Mixpanel project to which test analytics are sent."
  type        = "String"
  value       = "50ebae8c5fa7e5222b2271df8b73e91d"

  tags {
    Client      = "siberia"
    Environment = "test"
    Terraform   = "true"
  }
}
