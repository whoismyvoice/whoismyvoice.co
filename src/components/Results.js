import React from 'react';
import Settings from '../data/settings.json';
import HFCMembers from '../data/HFCMembers';

// Components
import Circle from './Circle';
import ArrowDown from './ArrowDown';
import SupportActions from './Senator/SupportActions';
import HFCOverview from './HFCOverview';
import FadedBG from './FadedBg';
import CongressmanGroup from './Senator/CongressmanGroup';
import BaseComponent from './BaseComponent';
import Button from './Button';

class Results extends BaseComponent {
	render() {
		const {
      vote_status,
      impact,
      current_member,
      zip_code,
      numRep,
      representatives,
      backgroundClasses,
      first_reps,
      initialize
    } = this.props,

    {no_cosponsor_title, no_cosponsor_desc} = Settings.senate,
    {chamber} = Settings;

    if (chamber === 'senate' && numRep === 0) {
      return <div className={backgroundClasses} id="fullpage">
        <FadedBG color="red" />
        <div className="section block two">
          <p className="impact">
            {no_cosponsor_title} <br />
            {no_cosponsor_desc}
          </p>
          <HFCOverview
            color="bright-red"
            members={HFCMembers}
          />
          <ArrowDown
            color="red-text"
            scroll={"true"}
          />
        </div>
      </div>;
    } else {
		  return <div className={backgroundClasses} id="fullpage">
        <div className="section block two">
          <Circle
            style="wide"
            desc={vote_status}
            numRep={numRep}
            representatives={representatives}
          />
          <CongressmanGroup
            representatives={representatives}
            zip_code={zip_code}
          />
        </div>
        <div className="section block three">
          <Circle
            style="wider"
            hide={true}
            desc={impact}
          />
          <SupportActions
            representatives={representatives}
            currentSenator={current_member}
          />
          <a
            className="internal-link"
            href="/about">
            About this project
          </a>

        </div>
      </div>;
    }
  }
};

Results.propTypes = {
  backgroundClasses: React.PropTypes.any,
  current_member: React.PropTypes.any,
  impact: React.PropTypes.string,
  numRep: React.PropTypes.number,
  representatives: React.PropTypes.array,
  vote_status: React.PropTypes.string,
  zip_code: React.PropTypes.string,
  initialize: React.PropTypes.func
};

export default Results;
