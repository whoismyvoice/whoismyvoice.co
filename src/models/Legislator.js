// @flow

import PropTypes from 'prop-types';
import { PropType as ContributionType } from './Contribution';

// Types
import type { ContributionRecord } from './Contribution';

export type BioguideId = string;
export type FecId = string;
opaque type LegislatorChamber = 'senate' | 'house' | 'unknown';
export type LegislatorIdentifier = string;

type LegislatorBioRecord = {
  birthday: string,
  gender: string,
};

export type LegislatorChannelRecord = {
  id: string,
  type: string,
};

type LegislatorIdRecord = {
  bioguide: BioguideId,
  fec: Array<FecId>,
};

type LegislatorNameRecord = {
  first?: string,
  last?: string,
  official_full: string,
};

type LegislatorTermRecord = {
  type: string,
  start: string,
  end: string,
  state: string,
  district?: string,
  party: string,
  phone: string,
};

export type LegislatorRecord = {
  id: LegislatorIdRecord,
  identifier?: string,
  name: string | LegislatorNameRecord,
  bio: LegislatorBioRecord,
  channels: Array<LegislatorChannelRecord>,
  photoUrl?: string,
  terms: Array<LegislatorTermRecord>,
};

/**
 * Defines the `PropTypes` validation structure for a `Legislator` record.
 */
export const PropType = PropTypes.shape({
  id: PropTypes.shape({
    bioguide: PropTypes.string,
    fec: PropTypes.arrayOf(PropTypes.string),
  }),
  identifier: PropTypes.string,
  name: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      first: PropTypes.string,
      last: PropTypes.string,
      official_full: PropTypes.string.isRequired,
    }),
  ]),
  bio: PropTypes.shape({
    birthday: PropTypes.string,
    gender: PropTypes.string,
  }),
  channels: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      type: PropTypes.string,
    })
  ),
  photoUrl: PropTypes.string,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      type: PropTypes.string.isRequired,
      start: PropTypes.string.isRequired,
      end: PropTypes.string.isRequired,
      state: PropTypes.string,
      district: PropTypes.number,
      party: PropTypes.string,
    })
  ),
});

/**
 * Remove accents and diacritic marks from the given string.
 * @param {string} str to clone without 'special' characters.
 * @returns `str` with the accents and diacritic characters replaced with their
 *    plain latin variants.
 */
function stripDiacritics(str) {
  return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
}

/**
 * Provides some convenience methods on top of the record for a legislator.
 */
export class Legislator {
  /**
   * Get the legislator's identifier from the given `record`.
   * @param {LegislatorRecord} record that matches the `PropType` defined for
   *    `Legislator` instances.
   * @returns the identifier used for legislator records for the given `record`.
   */
  static getIdentifier(record: LegislatorRecord): LegislatorIdentifier {
    PropTypes.checkPropTypes(
      { record: PropType },
      { record },
      'record',
      'Legislator#getIdentifier'
    );
    let identifier = undefined;
    if (record.identifier !== undefined) {
      identifier = record.identifier;
    } else if (typeof record.name === 'string') {
      identifier = stripDiacritics(record.name);
    } else {
      identifier = stripDiacritics(record.name.official_full);
    }
    return identifier;
  }

  /**
   * Retrieve the first contribution out of `contributions` that matches the identifier
   * of the given legislator.
   * @param {Array<ContributionRecord>} contributions records to be searched.
   * @param {LegislatorRecord|Legislator} legislator instance or record for which contributions
   *    will be searched.
   * @returns `-1` if no records from `contributions` match `legislator`, `amount`
   *    property of the contribution record otherwise.
   */
  static getContributionAmount(
    contributions: Array<ContributionRecord>,
    legislator: LegislatorRecord | Legislator
  ): number {
    PropTypes.checkPropTypes(
      { contributions: PropTypes.arrayOf(ContributionType) },
      { contributions },
      'contributions',
      'Legislator#getContributionAmount'
    );
    const id =
      legislator instanceof Legislator
        ? legislator.identifier
        : Legislator.getIdentifier(legislator);
    const contribution = contributions.find(
      record => record.legislatorId === id
    );
    if (contribution !== undefined) {
      return contribution.amount;
    } else {
      return -1;
    }
  }

  record: LegislatorRecord;

  /**
   * Create a new Legislator instance from the data provided in the given
   * `record`.
   *
   *      {
   *        "terms": [
   *          {
   *            "type": "rep|sen",
   *            "start": "<date string>",
   *            "end": "<date string>",
   *            "state": "<US State abbreviation>",
   *            "party": "<US Political Party Name>",
   *          }
   *        ]
   *      }
   *
   * @param {LegislatorRecord} record containing data about the legislator.
   */
  constructor(record: LegislatorRecord) {
    PropTypes.checkPropTypes(
      { record: PropType },
      { record },
      'record',
      'Legislator'
    );
    this.record = record;
  }

  /**
   * Gets the biographical object for the instance.
   * @returns an object containing (at least) `birthday` and `gender` properties.
   */
  get bio(): LegislatorBioRecord {
    return this.record.bio;
  }

  /**
   * Retrieve the bioguide property out of `record.id`.
   * @returns the bioguide id assigned to this legislator.
   */
  get bioguide(): BioguideId {
    return this.id.bioguide;
  }

  /**
   * Get the chamber of the legislator.
   * @returns 'house' if the legislator's current term is in the US House of
   *    Representatives; 'senate' if the legislator's current term is in the
   *    US Senate; 'unknown' otherwise.
   */
  get chamber(): LegislatorChamber {
    if (this.isSenator) {
      return 'senate';
    } else if (this.isRepresentative) {
      return 'house';
    } else {
      return 'unknown';
    }
  }

  /**
   * Retrieve the most recent term object from the given record.
   * @returns a term record for the most recent (current) term.
   */
  get currentTerm(): LegislatorTermRecord {
    const { terms } = this;
    const [currentTerm] = terms.slice(-1);
    return currentTerm;
  }

  /**
   * Retrieve the full, official name of the Legislator.
   * @returns full, official name.
   */
  get fullName(): string {
    return this.name.official_full;
  }

  /**
   * Retrieve the abbreviated gender of this legislator.
   * @returns the `gender` recorded for the legislator.
   */
  get gender(): string {
    return this.bio.gender;
  }

  /**
   * Retrieve a pronoun appropriate to the `gender` property.
   * @returns 'Them' if `gender` is anything other than 'M' or 'F'.
   */
  get genderPronoun(): string {
    switch (this.gender) {
      case 'M':
        return 'Him';
      case 'F':
        return 'Her';
      default:
        return 'Them';
    }
  }

  /**
   * Gets the id object for the instance.
   * @returns an object containing (at least) `fec` and `bioguide` properties.
   */
  get id(): LegislatorIdRecord {
    return this.record.id;
  }

  /**
   * Uses `Legislator#getIdentifier(record)` using the record provided at
   * instance creation time.
   * @returns the identifier used for legislator records for the given `record`.
   */
  get identifier(): LegislatorIdentifier {
    return Legislator.getIdentifier(this.record);
  }

  /**
   * Whether or not the `currentTerm` reflects a US Senate term.
   * @returns `true` if the `currentTerm` exists and has a Senate type.
   */
  get isSenator(): boolean {
    return this.currentTerm !== undefined && this.currentTerm.type === 'sen';
  }

  /**
   * Whether or not the `currentTerm` reflects a US House of Representatives
   * term.
   * @returns `true` if the `currentTerm` exists and has a US House of
   *    Representatives type.
   */
  get isRepresentative(): boolean {
    return this.currentTerm !== undefined && this.currentTerm.type === 'rep';
  }

  /**
   * Gets the name object for the instance.
   * @returns an object containing (at least) the `official_full` property.
   */
  get name(): LegislatorNameRecord {
    if (typeof this.record.name === 'string') {
      return { official_full: this.record.name };
    } else {
      return this.record.name;
    }
  }

  /**
   * Retrieves the US Political Party under which this legislator was elected
   * for their current term.
   * @returns current US Political Party.
   */
  get party(): string {
    return this.currentTerm.party;
  }

  /**
   * Get the phone number associated with the current term.
   * @returns The phone number associated with the current term.
   */
  get phone(): string {
    return this.currentTerm.phone;
  }

  /**
   * Get the URL for the official bioguide photo.
   * @returns a URL on theunitedstates.io.
   */
  get photoUrl(): string {
    const bioguide = this.bioguide;
    const size = 'original';
    return `https://theunitedstates.io/images/congress/${size}/${bioguide}.jpg`;
  }

  /**
   * The US State abbreviation for the state this legislator represents.
   * @returns a US State abbreviation.
   */
  get state(): string {
    return this.currentTerm.state;
  }

  /**
   * Gets the array of terms for the instance.
   * @returns an array of term records containing (at least) one record.
   */
  get terms(): Array<LegislatorTermRecord> {
    return this.record.terms;
  }

  /**
   * Get the Twitter handle of the legislator if it is provided.
   * @returns `undefined` if it isn't provided in `record`; the Twitter handle
   *    if it can be found in `record`.
   */
  get twitter(): ?string {
    const channels = this.record.channels || [];
    const channel = channels.find(channel => channel.type === 'Twitter');
    return channel === undefined ? undefined : channel.id;
  }
}

export default Legislator;
