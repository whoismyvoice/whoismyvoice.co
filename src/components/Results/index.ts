import { connect } from 'react-redux';
import { DispatchProps, StateProps, Results as Component } from './Results';
import { reset, navigateLegislators } from '../../actions';
import { Dispatch } from '../../actions/types';
import { State } from '../../store';

export const Results = Component;

function stateToProps(state: State): StateProps {
  const { address } = state;
  return {
    didSearch: address.value !== undefined,
    sectors: state.contributions.sectors,
  };
}

function dispatchToProps(dispatch: Dispatch): DispatchProps {
  function handleNavigation(event: React.MouseEvent<HTMLElement>): void {
    event.preventDefault();
    const target = event.currentTarget;
    if (target instanceof HTMLAnchorElement) {
      const href = target.href;
      if (href.endsWith('#section-0')) {
        dispatch(reset());
      } else if (href.endsWith('#section-1')) {
        dispatch(navigateLegislators(1));
      } else if (href.endsWith('#section-2')) {
        dispatch(navigateLegislators(2));
      }
    }
  }

  return {
    onBack: handleNavigation,
  };
}

const connector = connect(stateToProps, dispatchToProps);

export default connector(Results);
