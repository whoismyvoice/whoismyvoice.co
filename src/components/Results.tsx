import * as React from 'react';
import { connect } from 'react-redux';
import * as cx from 'classnames';

// Components
import TextButton from './Buttons/TextButton';
import MemberResults from './MemberResults';
import { Contribution } from '../models/Contribution';
import { Legislator, Record as LegislatorRecord } from '../models/Legislator';
import { reset } from '../actions';
import { Dispatch } from '../actions/types';
import { State } from '../store';

interface Props {
  contributions: Array<Contribution>;
  destroy?: Function;
  didSearch: boolean;
  representatives: Array<LegislatorRecord>;
  onBack?: Function;
}

export class Results extends React.Component<Props, {}> {
  static defaultProps = {
    didSearch: false,
    contributions: [],
    representatives: [],
  };

  getButtonProps(index: number) {
    const { onBack } = this.props;
    return index === 0 ? { onClick: onBack } : { link: `#section-${index}` };
  }

  render() {
    const { didSearch, contributions, representatives } = this.props;

    const backgroundClasses = cx(['second-wrapper'], {
      'move-up': didSearch,
    });

    const calcButtonProps = this.getButtonProps.bind(this);
    const legislators = representatives.map(rep => new Legislator(rep));
    const getAmount = Legislator.getContributionAmount.bind(
      this,
      contributions
    );
    const firstRep = legislators.filter(rep => getAmount(rep) > 0);
    const secondRep = legislators.filter(rep => getAmount(rep) === 0);
    const sections = [firstRep, secondRep]
      .filter(partition => partition.length > 0)
      .map((partition, index) => (
        <div
          key={partition.reduce(
            (key, legislator) => key + legislator.identifier,
            ''
          )}
          className="section block"
          id={`section-${index + 1}`}
        >
          <TextButton text="Back" {...calcButtonProps(index)} />
          <MemberResults
            legislators={partition}
            contributions={contributions}
            section={index + 1}
          />
        </div>
      ));

    return (
      <div className={backgroundClasses} id="fullpage">
        {sections}
      </div>
    );
  }
}

function stateToProps(state: State) {
  const { address, view } = state;
  return {
    didSearch: address.value !== undefined,
    ...view,
  };
}

function dispatchToProps(dispatch: Dispatch) {
  return {
    onBack: (event: Event) => {
      event.preventDefault();
      dispatch(reset());
    },
  };
}

export default connect<{}, {}, Props>(stateToProps, dispatchToProps)(Results);
