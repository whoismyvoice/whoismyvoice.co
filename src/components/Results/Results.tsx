import * as React from 'react';
import * as cx from 'classnames';

// Components
import TextButton from '../Buttons/TextButton';
import MemberResults from '../MemberResults';
import { Contribution } from '../../models/Contribution';
import {
  Legislator,
  Record as LegislatorRecord,
} from '../../models/Legislator';

export interface StateProps {
  didSearch?: boolean;
}

export interface DispatchProps {
  onBack?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
}

export interface Props extends StateProps, DispatchProps {
  contributions: Array<Contribution>;
  representatives: Array<LegislatorRecord>;
}

export class Results extends React.Component<Props> {
  static defaultProps = {
    didSearch: false,
    contributions: [],
    representatives: [],
    onBack: () => undefined,
  };

  getButtonProps(index: number) {
    const { onBack } = this.props;
    return index === 0 ? { onClick: onBack } : { link: `#section-${index}` };
  }

  render() {
    const { didSearch, contributions, representatives } = this.props;

    const backgroundClasses = cx(['second-wrapper'], {
      'move-up': didSearch,
    });

    const calcButtonProps = this.getButtonProps.bind(this);
    const legislators = representatives.map(rep => new Legislator(rep));
    const getAmount = Legislator.getContributionAmount.bind(
      this,
      contributions
    );
    const firstRep = legislators.filter(rep => getAmount(rep) > 0);
    const secondRep = legislators.filter(rep => getAmount(rep) === 0);
    const sections = [firstRep, secondRep]
      .filter(partition => partition.length > 0)
      .map((partition, index) => (
        <div
          key={partition.reduce(
            (key, legislator) => key + legislator.identifier,
            ''
          )}
          className="section block"
          id={`section-${index + 1}`}
        >
          <TextButton text="Back" {...calcButtonProps(index)} />
          <MemberResults
            legislators={partition}
            contributions={contributions}
            section={index + 1}
          />
        </div>
      ));

    return (
      <div className={backgroundClasses} id="fullpage">
        {sections}
      </div>
    );
  }
}
