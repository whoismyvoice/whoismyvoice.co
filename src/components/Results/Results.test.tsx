import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Results } from './Results';
import { createLegislator, createTerm } from '../../models/Legislator.test';

it('renders correctly with empty props', () => {
  const props = {
    representatives: [],
    sectorContributions: {},
  };
  const { container, queryByText, queryByTestId } = render(
    <Results {...props} />
  );
  expect(container).toBeInTheDocument();
  expect(queryByTestId('result-section-1')).toBeNull();
  expect(queryByText('Search Another Zip Code')).toBeNull();
});

it('renders correctly with one legislator', () => {
  const props = {
    representatives: [createLegislator('John Smith')],
    sectorContributions: {},
  };
  const { container, getByTestId, getByText, queryByTestId } = render(
    <Results {...props} />
  );
  expect(container).toBeInTheDocument();
  expect(getByTestId('result-section-1')).toBeInTheDocument();
  expect(queryByTestId('result-section-2')).toBeNull();
  expect(getByText('Search Another Zip Code')).toBeInTheDocument();
});

it('renders correctly with two legislators', () => {
  const props = {
    representatives: [
      createLegislator('John Smith'),
      createLegislator('John Smith Jr.'),
    ],
    sectorContributions: {},
  };
  const { container, getByTestId, getByText, queryByTestId } = render(
    <Results {...props} />
  );
  expect(container).toBeInTheDocument();
  expect(getByTestId('result-section-1')).toBeInTheDocument();
  expect(queryByTestId('result-section-2')).toBeNull();
  expect(getByText('Search Another Zip Code')).toBeInTheDocument();
});

it('renders correctly with three legislators', () => {
  const props = {
    representatives: [
      createLegislator('John Smith'),
      createLegislator('John Smith Jr'),
      { ...createLegislator('John Smith III'), terms: [createTerm('rep')] },
    ],
    sectorContributions: {},
  };
  const { container, getByTestId, getByText } = render(<Results {...props} />);
  expect(container).toBeInTheDocument();
  expect(getByTestId('result-section-1')).toBeInTheDocument();
  expect(getByTestId('result-section-2')).toBeInTheDocument();
  expect(getByText('Search Another Zip Code')).toBeInTheDocument();
});
