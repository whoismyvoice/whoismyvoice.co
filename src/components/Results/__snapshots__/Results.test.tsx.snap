// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with contribution and legislator 1`] = `
<div
  className="second-wrapper"
  id="fullpage"
>
  <div
    className="section block"
    id="section-1"
  >
    <a
      className="text-button text-button--back"
      href={undefined}
      onClick={[Function]}
    >
      Back
      <span
        className="text-button-border"
      />
    </a>
    <span>
      <span>
        <div
          className="title-component title-component--results"
        >
          <div
            className="title-component__description"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Your Senator <span class=\\"bold\\"><b>accepted money</b></span> <span class=\\"strike-out\\">from the NRA</span> in their recent election cycles.",
              }
            }
          />
        </div>
        <div
          className="member-wrapper"
        >
          <div
            className="member-container"
          >
            <div
              className="member-img member--grey"
            >
              <img
                alt="John Smith"
                src="https://theunitedstates.io/images/congress/original/.jpg"
              />
            </div>
            <div
              className="ribbon-wrapper"
            >
              <h1
                className="ribbon ribbon--grey"
              >
                <strong
                  className="ribbon-content"
                >
                  (
                  I
                   - 
                  NY
                  ) 
                  John Smith
                </strong>
              </h1>
            </div>
            <span
              className="member__payment"
            >
              <span
                className="counter__accept"
              >
                Accepted
              </span>
              <span
                className="counter__digit digit__margin"
              >
                $
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                1
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__from"
              >
                from 
                the NRA
              </span>
            </span>
            <div
              className="mobile-contact-options"
              id="legislator-contact-0"
            >
              Contact Them
            </div>
            <div
              className="actionButtons"
            >
              <a
                href="tel:123-123-1234"
                target="_blank"
              >
                <button
                  className="contact__button_small undefined"
                >
                  <div
                    className="button__small__icon phone"
                  />
                  <div
                    className="button__small__text"
                  >
                    Call Them
                  </div>
                </button>
              </a>
            </div>
          </div>
        </div>
      </span>
      <span>
        <div
          className="line-seperator line-seperator--small"
        />
        <a
          className="text-button"
          href="#section-2"
          onClick={[Function]}
        >
          See My Other Representatives
          <span
            className="text-button-border"
          />
        </a>
      </span>
    </span>
  </div>
</div>
`;

exports[`renders correctly with empty props 1`] = `
<div
  className="second-wrapper"
  id="fullpage"
/>
`;

exports[`renders correctly with three contributions and legislators 1`] = `
<div
  className="second-wrapper"
  id="fullpage"
>
  <div
    className="section block"
    id="section-1"
  >
    <a
      className="text-button text-button--back"
      href={undefined}
      onClick={[Function]}
    >
      Back
      <span
        className="text-button-border"
      />
    </a>
    <span>
      <span>
        <div
          className="title-component title-component--results"
        >
          <div
            className="title-component__description"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Your Senators & Representative <span class=\\"bold\\"><b>accepted money</b></span> <span class=\\"strike-out\\">from the NRA</span> in their recent election cycles.",
              }
            }
          />
        </div>
        <div
          className="member-wrapper"
        >
          <div
            className="member-container"
          >
            <div
              className="member-img member--grey"
            >
              <img
                alt="John Smith"
                src="https://theunitedstates.io/images/congress/original/.jpg"
              />
            </div>
            <div
              className="ribbon-wrapper"
            >
              <h1
                className="ribbon ribbon--grey"
              >
                <strong
                  className="ribbon-content"
                >
                  (
                  I
                   - 
                  NY
                  ) 
                  John Smith
                </strong>
              </h1>
            </div>
            <span
              className="member__payment"
            >
              <span
                className="counter__accept"
              >
                Accepted
              </span>
              <span
                className="counter__digit digit__margin"
              >
                $
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                1
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__from"
              >
                from 
                the NRA
              </span>
            </span>
            <div
              className="mobile-contact-options"
              id="legislator-contact-0"
            >
              Contact Them
            </div>
            <div
              className="actionButtons"
            >
              <a
                href="tel:123-123-1234"
                target="_blank"
              >
                <button
                  className="contact__button_small undefined"
                >
                  <div
                    className="button__small__icon phone"
                  />
                  <div
                    className="button__small__text"
                  >
                    Call Them
                  </div>
                </button>
              </a>
            </div>
          </div>
          <div
            className="member-container"
          >
            <div
              className="member-img member--grey"
            >
              <img
                alt="John Smith Jr."
                src="https://theunitedstates.io/images/congress/original/.jpg"
              />
            </div>
            <div
              className="ribbon-wrapper"
            >
              <h1
                className="ribbon ribbon--grey"
              >
                <strong
                  className="ribbon-content"
                >
                  (
                  I
                   - 
                  NY
                  ) 
                  John Smith Jr.
                </strong>
              </h1>
            </div>
            <span
              className="member__payment"
            >
              <span
                className="counter__accept"
              >
                Accepted
              </span>
              <span
                className="counter__digit digit__margin"
              >
                $
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                1
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__from"
              >
                from 
                the NRA
              </span>
            </span>
            <div
              className="mobile-contact-options"
              id="legislator-contact-1"
            >
              Contact Them
            </div>
            <div
              className="actionButtons"
            >
              <a
                href="tel:123-123-1234"
                target="_blank"
              >
                <button
                  className="contact__button_small undefined"
                >
                  <div
                    className="button__small__icon phone"
                  />
                  <div
                    className="button__small__text"
                  >
                    Call Them
                  </div>
                </button>
              </a>
            </div>
          </div>
          <div
            className="member-container"
          >
            <div
              className="member-img animated member--grey"
            >
              <img
                alt="John Smith III"
                src="https://theunitedstates.io/images/congress/original/.jpg"
              />
            </div>
            <div
              className="ribbon-wrapper"
            >
              <h1
                className="ribbon ribbon--grey"
              >
                <strong
                  className="ribbon-content"
                >
                  (
                  I
                   - 
                  NY
                  ) 
                  John Smith III
                </strong>
              </h1>
            </div>
            <span
              className="member__payment"
            >
              <span
                className="counter__accept"
              >
                Accepted
              </span>
              <span
                className="counter__digit digit__margin"
              >
                $
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                1
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__from"
              >
                from 
                the NRA
              </span>
            </span>
            <div
              className="mobile-contact-options"
              id="legislator-contact-2"
            >
              Contact Them
            </div>
            <div
              className="actionButtons"
            >
              <a
                href="tel:123-123-1234"
                target="_blank"
              >
                <button
                  className="contact__button_small undefined"
                >
                  <div
                    className="button__small__icon phone"
                  />
                  <div
                    className="button__small__text"
                  >
                    Call Them
                  </div>
                </button>
              </a>
            </div>
          </div>
        </div>
      </span>
      
    </span>
  </div>
</div>
`;

exports[`renders correctly with two contributions and legislators 1`] = `
<div
  className="second-wrapper"
  id="fullpage"
>
  <div
    className="section block"
    id="section-1"
  >
    <a
      className="text-button text-button--back"
      href={undefined}
      onClick={[Function]}
    >
      Back
      <span
        className="text-button-border"
      />
    </a>
    <span>
      <span>
        <div
          className="title-component title-component--results"
        >
          <div
            className="title-component__description"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Your Senators <span class=\\"bold\\"><b>accepted money</b></span> <span class=\\"strike-out\\">from the NRA</span> in their recent election cycles.",
              }
            }
          />
        </div>
        <div
          className="member-wrapper"
        >
          <div
            className="member-container"
          >
            <div
              className="member-img member--grey"
            >
              <img
                alt="John Smith"
                src="https://theunitedstates.io/images/congress/original/.jpg"
              />
            </div>
            <div
              className="ribbon-wrapper"
            >
              <h1
                className="ribbon ribbon--grey"
              >
                <strong
                  className="ribbon-content"
                >
                  (
                  I
                   - 
                  NY
                  ) 
                  John Smith
                </strong>
              </h1>
            </div>
            <span
              className="member__payment"
            >
              <span
                className="counter__accept"
              >
                Accepted
              </span>
              <span
                className="counter__digit digit__margin"
              >
                $
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                1
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__from"
              >
                from 
                the NRA
              </span>
            </span>
            <div
              className="mobile-contact-options"
              id="legislator-contact-0"
            >
              Contact Them
            </div>
            <div
              className="actionButtons"
            >
              <a
                href="tel:123-123-1234"
                target="_blank"
              >
                <button
                  className="contact__button_small undefined"
                >
                  <div
                    className="button__small__icon phone"
                  />
                  <div
                    className="button__small__text"
                  >
                    Call Them
                  </div>
                </button>
              </a>
            </div>
          </div>
          <div
            className="member-container"
          >
            <div
              className="member-img member--grey"
            >
              <img
                alt="John Smith Jr."
                src="https://theunitedstates.io/images/congress/original/.jpg"
              />
            </div>
            <div
              className="ribbon-wrapper"
            >
              <h1
                className="ribbon ribbon--grey"
              >
                <strong
                  className="ribbon-content"
                >
                  (
                  I
                   - 
                  NY
                  ) 
                  John Smith Jr.
                </strong>
              </h1>
            </div>
            <span
              className="member__payment"
            >
              <span
                className="counter__accept"
              >
                Accepted
              </span>
              <span
                className="counter__digit digit__margin"
              >
                $
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                1
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__digit"
              >
                0
                <span
                  className="counter__gradient-top"
                />
              </span>
              <span
                className="counter__from"
              >
                from 
                the NRA
              </span>
            </span>
            <div
              className="mobile-contact-options"
              id="legislator-contact-1"
            >
              Contact Them
            </div>
            <div
              className="actionButtons"
            >
              <a
                href="tel:123-123-1234"
                target="_blank"
              >
                <button
                  className="contact__button_small undefined"
                >
                  <div
                    className="button__small__icon phone"
                  />
                  <div
                    className="button__small__text"
                  >
                    Call Them
                  </div>
                </button>
              </a>
            </div>
          </div>
        </div>
      </span>
      <span>
        <div
          className="line-seperator line-seperator--small"
        />
        <a
          className="text-button"
          href="#section-2"
          onClick={[Function]}
        >
          See My Representative
          <span
            className="text-button-border"
          />
        </a>
      </span>
    </span>
  </div>
</div>
`;
